src/runtime.js

```diff
@@ -266,9 +272,14 @@ function adoptShared(global, opts, callback) {
    */
   global.AMP.push = function(fnOrStruct) {
     // Extensions are only processed once HEAD is complete.
-    waitForBody(global.document, () => {
-      installExtension(fnOrStruct);
-    });
+    const register = function() {
+      waitForBody(global.document, () => {
```

Comment:
It might be better to move `waitForBody` outside. It's a very rare thing that body is not available by this time.